### PR TITLE
Allow model rendering code to use polymodel_instance. 

### PR DESCRIPTION
**Pull request by SamuelCho**
Removes the reliance on ship_model_start/stop for rendering models. Also fix a bunch of oversights involving blown off models in the polymodel_instance code.

### CODE REVIEW

File: code/model/modelread.cpp [hunk 1]
```diff
@@ -4547,11 +4547,14 @@ void model_clear_instance_info( submodel_instance_info * sii )
...
-void model_clear_submodel_instance( submodel_instance *sm_instance )
+void model_clear_submodel_instance( submodel_instance *sm_instance, bsp_info *sm )
 {
 	sm_instance->angs.p = 0.0f;
 	sm_instance->angs.b = 0.0f;
 	sm_instance->angs.h = 0.0f;
+
+	sm_instance->blown_off = sm->is_damaged ? true : false;
+
 	sm_instance->blown_off = false;
```
Comment (MageKing17) [inline code/model/modelread.cpp]: Was this line supposed to be removed?

**Comment by SamuelCho** [comment]
Yes, that line was intentionally replaced. Blown_off needs to match the bsp_info's damage state.

**Comment by The-E** [comment]
Yeah, okay, but why set that variable twice, once to what I assume is a correct value and then to false immediately after that?

**Comment by The-E** [comment]
Point is, either line 4556 or line 4558 should be removed.

**Comment by MageKing17** [comment]
I was inquiring whether or not you intended to remove the line, not asking why you did, because... you didn't.

**Comment by SamuelCho** [comment]
The line below is supposed to be removed. I thought I saw `-` marks on that line because I'm dyslexic and shouldn't be a programmer.

**Comment by SamuelCho** [comment]
It's fixed.

**Pull request merged**